cli/command/cli.go

```diff
@@ -145,13 +147,38 @@ func getConfiguredCredentialStore(c *configfile.ConfigFile, serverAddress string
 	return c.CredentialsStore
 }
 
+func newAPIClientWithTLSKeyPass(opts *cliflags.CommonOptions, configFile *configfile.ConfigFile, passRetriever notary.PassRetriever) (client.APIClient, error) {
+	client, err := NewAPIClientFromFlags(opts, configFile)
+	if err == nil {
+		return client, nil
+	}
+
+	var (
+		passwd string
+		giveup bool
+	)
+	for attempts := 0; tlsconfig.IsErrEncryptedKey(err); attempts++ {
+		// some code and comments borrowed from notary/trustmanager/keystore.go
+		passwd, giveup, err = passRetriever("private", "encrypted TLS private", false, attempts)
+		// Check if the passphrase retriever got an error or if it is telling us to give up
+		if giveup || err != nil {
+			return nil, fmt.Errorf("private key is encrypted, but could not get passphrase: %v", err)
```

Comment:
Ooooh, I'm going to have to look at that package for Notary (cc @cyli @riyazdf: it adds stack traces to errors)